### PR TITLE
Updated Docs For Java

### PR DESCRIPTION
**Pull request by gldeA**
I updated some of the docs files from last year, editing them and converting C++ to Java.

### CODE REVIEW

File: docs/command_based.md [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+
+1. Add a new parameter to your command's constructor of type `RETURNTYPESupplier`, where `RETURNTYPE` is whatever type your lambda will return.
+    - You will need to import the supplier from `java.util.function.RETURNTYPESupplier`
+    - For example, a joystick axis has the type `double`, so your parameter would be of type `DoubleSupplier` and you would import `java.util.function.DoubleSupplier`
+2. Make a variable of that same type at the top of your command's class so that the supplier is accessible after the constructor (`DoubleSupplier x;`)
+    - Remember to set your member variable equal to your parameter in the constructor
+3. When you bind your command, you can use the following format to define your lambda:
+    ```java
+    controller.y().whileTrue(new Shoot(shooter, 
+    () -> { return controller.getLeftY(); }));
```
[the remark this answers is not in the document]
Comment (cfinn68916) [reply docs/command_based.md:234]: Or just pass in the whole XboxController object

**Review by cfinn68916 — CHANGES_REQUESTED** [review]
See comments

File: docs/command_based.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (gldeA) [reply docs/command_based.md:234]: Huh, I did not know about the :: operator in Java...

File: docs/wpilib_reference.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+4. [Posting NetworkTables Values](#posting-networktables-values)
+5. [Solenoids](#solenoids)
+
+## Component Class Reference:
+
+|Class|Import Path|Notes|
+|-|-|-|
+|`TalonFX`|`com.ctre.phoenix6.hardware.TalonFX`|-|
+|`CANSparkMax`|`com.revrobotics.CANSparkMax`|-|
+|`TalonSRX`|`com.ctre.phoenix6.hardware.TalonSRX`|This is probably right, unconfirmed|
```
[the remark this answers is not in the document]
Comment (gldeA) [reply docs/wpilib_reference.md:17]: Ok... What's the correct path then?

[the remark this answers is not in the document]
Comment (gldeA) [reply docs/wpilib_reference.md:17]: `com.ctre.phoenix5.hardware.TalonSRX`?

File: docs/can_ids.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+Each subsystem on the robot gets 25 IDs, with 0-10 being motors, 11-15 being non-motor output, 16-20 being input, and 21-25 being miscellaneous. There are breaks in the table between each section.
+
+## Drivetrain
+
+|Component|ID|
+|-|-|
+|Front Left Motor|0|
+|Front Right Motor|1|
+|Back Left Motor|2|
```
[the remark this answers is not in the document]
Comment (gldeA) [reply docs/can_ids.md:11]: This is a mess

[the remark this answers is not in the document]
Comment (gldeA) [reply docs/can_ids.md]: Also, 10 and 20 were conflicting, so I changed them to 18 and 19

**Comment by gldeA** [comment]
Should be fixed

**Review by cfinn68916 — APPROVED** [review]
LGTM (let's gamble try merging). I still prefer passing the controller object reference into commands instead of lambdas.

**Comment by gldeA** [comment]
> I still prefer passing the controller object reference into commands instead of lambdas.
I disagree, best practice is generally to provide only as much access as you need, and passing the full controller in is unnecessary (though it could be a good idea when you need a lot of controller inputs in one command rather than just one or two)

**Pull request merged**